src/action/file_stat.rs

```diff
@@ -0,0 +1,532 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the file stat action.
+//!
+//! A file stat action responses with stat of a given file
+
+use std::fs::{self, File, Metadata};
+use std::path::{Path, PathBuf};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use log::warn;
+use rrg_proto::{GetFileStatRequest, path_spec::Options, path_spec::PathType,
+                stat_entry::ExtAttr, StatEntry};
+
+use crate::session::{self, Error, Session};
+
+impl From<std::io::Error> for Error {
+    fn from(e: std::io::Error) -> Error {
```

Comment:
I tend to have a blank newline between `impl`, `struct` and alike and the first item.

Reply:
Fixed.